src/generic_io.jl

```diff
@@ -12,13 +12,38 @@ const datatype_regexp = r"""^(([A-Za-z_]*)(<([0-9,]*)>)?)(\{(.*)\})?$"""
 const arraydims_regexp = r"""^<([0-9,]*)>$"""
 
 function _eldatatype_from_string(s::Union{Nothing,AbstractString})
-    if s == nothing || s == ""
+    if isnothing(s) || isempty(s)
         RealQuantity
     else
         datatype_from_string(s)
     end
 end
 
+dataselector_byname = Dict{String, DataType}(
+    "expsetup" => ExpSetup,
+    "datatier" => DataTier,
+    "datapartition" => DataPartition,
+    "dataperiod" => DataPeriod,
+    "datarun" => DataRun,
+    "datacategory" => DataCategory,
+    "timestamp" => Timestamp,
+    "filekey" => FileKey,
+    "channelid" => ChannelId,
+    "detectorid" => DetectorId
+)
+
+dataselector_bytypes = Dict{DataType, String}(
+    ExpSetup => "expsetup",
+    DataTier => "datatier",
+    DataPartition => "datapartition",
+    DataPeriod => "dataperiod",
+    DataRun => "datarun",
+    DataCategory => "datacategory",
+    Timestamp => "timestamp",
+    FileKey => "filekey",
+    ChannelId => "channelid",
+    DetectorId => "detectorid"
+)
```

Comment:
Maybe set the minimum version of LegendDataManagement to the version where all of these are already implemented

Reply:
LegendHDF5IO should not have a hard dependency on LegendDataManagement. This can be done via a soft dependency (extension) though.